src/display.c

```diff
@@ -139,16 +139,22 @@ plot_print_y_label(struct plot *p, struct canvas_elem e, double l, int side)
 
 	pp = side == 1 ? PPTLeft | ((e.peice & 0x8) >> 2) : PPTRight | e.peice;
 
+	if (side == 1) {
+		printf("\033[0m");
+		printf(p->y_label.l_fmt, l);
+	}
+
 	if (pp == PPCross && e.color > 0) {
 		printf("\033[%dm", e.color);
-	}else if (p->color) {
+	} else if (p->color) {
 		printf("\033[0m");
 	}
 
-	if (side == 1) {
-		printf(p->y_label.l_fmt, l, plot_charsets[p->charset][pp]);
-	}else if (side == 2) {
-		printf(p->y_label.r_fmt, plot_charsets[p->charset][pp], l);
+	printf("%s", plot_charsets[p->charset][pp]);
+
+	if (side == 2) {
+		printf("\033[0m");
```

Comment:
this is redundant, line 150 already clears the color

Reply:
I disagree on this one, line 150 does not reset the right side color (as it is a cross).

Test: `seq 1 100 | shuf | ./build/plot -c r -y 10:3:3`

Reply:
Ah yes, I see you are right.  Well, you should add the same `if (p->color) {` here then too.  I will just do it in a post-merge commit if there is nothing else.